### PR TITLE
[Feature] Make common_on_attach configurable

### PR DESCRIPTION
**Pull request by kylo252**
# Description

Allow the user to set a custom on_attach callback for lspconfig. 

This also fixes a bug where `lspconfig.common_on_attach` is undefined when `O.lsp.document_highlight` is set to `false`, but it was still called regardless.

## How Has This Been Tested?

Add this to `lv-config.lua` and verify that it's working
```lua
O.lsp.on_attach_callback = function(client, bufnr)
  -- ref: https://github.com/neovim/nvim-lspconfig#keybindings-and-completion
  local function buf_set_option(...)
    vim.api.nvim_buf_set_option(bufnr, ...)
  end
  --Enable completion triggered by <c-x><c-o>
  buf_set_option("omnifunc", "v:lua.vim.lsp.omnifunc")
end
```

### CODE REVIEW

File: lua/lsp/init.lua [hunk 1]
```diff
@@ -111,7 +111,10 @@ autocmd BufWritePre *.lua lua vim.lsp.buf.formatting_sync(nil, 100) ]]
 -- Java
 -- autocmd FileType java nnoremap ca <Cmd>lua require('jdtls').code_action()<CR>
 
-local function documentHighlight(client, bufnr)
+local function get_lsp_document_highlighting(client, bufnr)
```
Comment (Tastyep) [inline lua/lsp/init.lua:114]: How about `lsp_highlight_document` or `highlight_document` ?
My problem with this one is that it always returns nil, so it doesn't really get something.

File: lua/lsp/init.lua [hunk 2]
```diff
@@ -196,10 +199,11 @@ function lsp_config.PeekImplementation()
   end
 end
 
-if O.lsp.document_highlight then
-  function lsp_config.common_on_attach(client, bufnr)
-    documentHighlight(client, bufnr)
+function lsp_config.common_on_attach(client, bufnr)
+  if O.lsp.on_attach_callback ~= nil then
```
Comment (Tastyep) [inline lua/lsp/init.lua:203]: Can be simplified to `if O.lsp.on_attach_callback then`

**Review by Tastyep — CHANGES_REQUESTED** [review]
Looking good :+1:

**Review by Tastyep — APPROVED** [review]

File: utils/installer/lv-config.example-no-ts.lua [hunk 3]
```diff
@@ -50,6 +50,17 @@ O.treesitter.ensure_installed = {}
 O.treesitter.ignore_install = { "haskell" }
 O.treesitter.highlight.enabled = true
 
+-- generic LSP settings
+-- you can set a custom on_attach function that will be used for all the language servers
+-- See <https://github.com/neovim/nvim-lspconfig#keybindings-and-completion>
+O.lsp.on_attach_callback = function(client, bufnr)
```
Comment (abzcoding) [inline utils/installer/lv-config.example-no-ts.lua:56]: can we have these lines commented sir :pleading_face:

File: utils/installer/lv-config.example.lua [hunk 4]
```diff
@@ -50,6 +50,17 @@ O.treesitter.ensure_installed = "maintained"
 O.treesitter.ignore_install = { "haskell" }
 O.treesitter.highlight.enabled = true
 
+-- generic LSP settings
+-- you can set a custom on_attach function that will be used for all the language servers
+-- See <https://github.com/neovim/nvim-lspconfig#keybindings-and-completion>
+O.lsp.on_attach_callback = function(client, bufnr)
```
Comment (abzcoding) [inline utils/installer/lv-config.example.lua:56]: also these lines

File: README.md [hunk 5]
```diff
@@ -99,6 +99,16 @@ O.treesitter.ensure_installed = "all"
 O.treesitter.ignore_install = {"haskell"}
 O.treesitter.highlight.enabled = true
 
+-- you can set a custom on_attach function that will be used for all the language servers
+-- See <https://github.com/neovim/nvim-lspconfig#keybindings-and-completion>
+O.lsp.on_attach_callback = function(client, bufnr)
```
Comment (abzcoding) [inline README.md:104]: and possibly these lines

**Pull request merged**